packages/inertia-vue/src/form.js

```diff
@@ -0,0 +1,108 @@
+import Vue from 'vue'
+import { Inertia } from '@inertiajs/inertia'
+
+export default function(data) {
+  const defaults = JSON.parse(JSON.stringify(data))
+
+  return {
+    errors: {},
+    hasErrors: false,
+    processing: false,
+    data() {
+      return Object
+        .keys(data)
+        .reduce((carry, key) => {
+          carry[key] = this[key]
+          return carry
+        }, {})
+    },
+    submit(method, url, options) {
+      url = url || Inertia.page.url
+      options = options || {}
+
+      Inertia[method](url, this.data(), {
+        ... options,
+        onStart: visit => {
+          this.processing = true
+
+          if (options.onStart) {
+            return options.onStart(visit)
+          }
+        },
+        onFinish: () => {
+          this.processing = false
+
+          if (options.onFinish) {
+            return options.onFinish()
+          }
+        },
+        onError: errors => {
+          Vue.set(this, 'errors', errors)
+          this.hasErrors = true
+
+          if (options.onError) {
+            return options.onError(errors)
+          }
+        },
+        onSuccess: page => {
+          Vue.set(this, 'errors', {})
+          this.hasErrors = false
```

Comment:
This can use the `clearErrors` function.

```suggestion
          this.clearErrors()
```

Reply:
Good catch! 👍